### DISCUSSION TITLE
Why does the JavaScript generated by tsup use "var" no matter what target is specified in my tsconfig?

### DISCUSSION THREAD

**Discussion by kevincam3** [egoist/tsup#1228] [General]
My target in tsconfig is "target": "ES2022" and "module":"Node16". In these versions of JavaScript and Node both "const" and "let" were available. So I'm trying to understand why the generated JavaScript only uses "var"

**Comment by rtritto** [comment]
Seems that `const` and `let` are only used inside functions.
All `const` and `let` at first scope level are replaced with `var`.
The issue is related to the tree-shaking of `esbuild` that doesn't work with `target` (e.g. 'es2020' or 'esnext') option.

Related:
- https://esbuild.github.io/faq/#top-level-var
- https://github.com/evanw/esbuild/issues/623

PS: as alternative, I'm using `pkgroll`
